staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -473,6 +474,7 @@ func (w *watchCache) waitUntilFreshAndBlock(ctx context.Context, resourceVersion
 		w.cond.Wait()
 	}
 	span.AddEvent("watchCache fresh enough")
+	metrics.WatchCacheReadWait.WithContext(ctx).WithLabelValues(strconv.FormatUint(resourceVersion, 10)).Observe(w.clock.Since(startTime).Seconds())
```

Comment:
Think we need to include the timeouts on the line 472 too. Can you move it to the begining of the function and use defer?
```
func (w *watchCache) waitUntilFreshAndBlock(ctx context.Context, resourceVersion uint64) error {
	startTime := w.clock.Now()
	defer func() {
		metrics.WatchCacheReadWait.WithContext(ctx).Observe(w.clock.Since(startTime).Seconds())
	}()
```

Reply:
Additional note, not sure why we call this code when resourceVersion="0", as the wait no wait needed. I think we can not emit the metric if for `resourceVersion="0"` as it would just introduce noise. We can use the check `if resourceVersion>0 {